validate.py

```diff
@@ -0,0 +1,139 @@
+        'description': {'type': 'string'},
+    },
+    'required': ['name', 'description']
+}
+
+for application_dir in application_dirs:
+    application_json_file = application_dir + '/application.json'
+    try:
+        with open(application_json_file) as f:
+            application_json = json.loads(f.read().decode("utf-8-sig"))
+            jsonschema.validate(application_json, application_schema)
+    except ValueError as value_error:
+        print "invalid application.json {}: {}".format(application_json_file, value_error)
+        is_valid = False
+    except jsonschema.exceptions.ValidationError as validation_error:
+        print "invalid application.json {}: {}".format(application_json_file, validation_error)
+        is_valid = False
+
+# get script directories
+script_dirs = []
+for application_dir in application_dirs:
+    script_dirs_for_app = []
+    for script in next(os.walk(application_dir))[1]:
+        if os.path.isfile(application_dir + '/' + script + '/script.json') \
+                and os.path.isfile(application_dir + '/' + script + '/script.js'):
+            script_dirs_for_app.append(application_dir + '/' + script)
+    if script_dirs_for_app:
+        script_dirs.extend(script_dirs_for_app)
+    else:
+        print "application {} must contain at least one script directory with 'script.js' and 'script.json'".format(
+            application_dir)
+        is_valid = False
+
+# validate script.json
+script_schema = {
+    'type': 'object',
+    'properties': {
+        'scriptName': {'type': 'string'},
+        'compatibleOperatingSystems': {
+            'type': 'array',
+            'items': {'enum': ['LINUX', 'MACOSX']},
+            'maxItems': 2
+        },
+        'testingOperatingSystems': {
+            'type': 'array',
+            'items': {'enum': ['LINUX', 'MACOSX']},
+            'maxItems': 2
+        },
+        'free': {'type': 'boolean'},
+        'requiresPatch': {'type': 'boolean'},
+    },
+    'required': ['scriptName', 'compatibleOperatingSystems', 'testingOperatingSystems', 'free', 'requiresPatch']
+}
+
+for script_dir in script_dirs:
+    script_json_file = script_dir + '/script.json'
+    try:
+        with open(script_json_file) as f:
+            script_json = json.loads(f.read().decode("utf-8-sig"))
+            jsonschema.validate(script_json, script_schema)
+    except ValueError as value_error:
+        print "invalid script.json {}: {}".format(script_json_file, value_error)
+        is_valid = False
+    except jsonschema.exceptions.ValidationError as validation_error:
+        print "invalid script.json {}: {}".format(script_json_file, validation_error)
+        is_valid = False
+
+# check miniature
+for application_dir in application_dirs:
+    main_miniature_file = application_dir + '/miniatures/main.png'
+    if os.path.isfile(main_miniature_file):
+        image = Image.open(main_miniature_file)
+        width, height = image.size
+        if width != 400 or height != 300:
```

Comment:
Are you sure about that the dimensions are required?
As far as I remember, POL 5 will automatically scale the image to fit inside the correct region.
Depending on the type of selected list view this also retains the width/height proportions.
Therefore I don't think we need to validate the image dimensions.
In addition I believe we have some scripts that use slightly different dimensions. (I believe I added such an image myself a few months ago)

Reply:
We should require the dimensions otherwise the scaled image might look very strange (even if the proportions are preserved). 
For example, an image of 40x30px will be very blurry. An image of 4000x3000px might not be recognizable because it becomes too small. If the image has different proportions than 4:3, it will look strange in the "Applications" view (e.g. 200x200px).

From what I saw, there are currently only 3 main.png with wrong dimensions (one of those with 399x299px). I will leave them as a test for Travis and then fix them.